### PR TITLE
Next iteration for SSO

### PR DESCRIPTION
**Pull request by HannahSMead**

### CODE REVIEW

File: controls.py [hunk 1]
```diff
@@ -155,35 +112,6 @@ def request_info_who(post_n,type_n):
     return data
 # print(request_info()[1][0])
```
Comment (flllra) [inline controls.py:113]: maybe I am missing it but do we not have request_info() anymore? If not, how are we referencing db now?

File: controls.py [hunk 2]
```diff
@@ -5,64 +5,57 @@
...
-    s.add(Stu)
     s.commit()
     s.close()
+
     print(username, "added to the database")
 
-def create_donor(username, password, email, token=""):
-    # Create a Donor and Account instance in the rd database
+def approve_admin(usernameAttempt):
     s = db.get_session()
```
Comment (flllra) [inline controls.py:20]: I think we also have an ADMIN dictionary in config.py, maybe we should have it so admins are added to that list in config as well?

File: controls.py [hunk 3]
```diff
@@ -5,64 +5,57 @@
...
+    A.approved = True
+    s.commit()
+    s.close()
 
-def open_request(student_id, amount, reason, title="Donate to a Reedie in needie", anon=True, amount_filled=0.0):
+def open_request(student_id, amount, reason, title, anon):
     # Create a request instance in the rd database.
     s = db.get_session()
-    req = Request(requested_by=student_id, amount_needed=amount, title=title, amount_filled=amount_filled, description=reason, anon=anon)
+    if title == '': title = "Donate to a Reedie in needie"
```
Comment (flllra) [inline controls.py:36]: I feel like we should just make it so the "title" input on "make a request" is required?

File: controls.py [hunk 4]
```diff
@@ -5,64 +5,57 @@
...
-    D.money_given += amount
-    D.donations_given += 1
+    A.given += amount
+    A.num_given += 1
+    s.flush()
     R.amount_filled += amount
 
     # check to see if the request has been completed
     if R.amount_needed >= R.amount_filled:
         R.filled = True
```
Comment (flllra) [inline controls.py:56]: is this where we should start referencing stripe data? or do we even need to do that really?

File: schema.py [hunk 5]
```diff
@@ -33,54 +33,41 @@ class Account(Base, IdPrimaryKeyMixin, DateTimeMixin):
...
+    admin = Column(Boolean, default=False) #True for admins
+    approved = Column(Boolean, default=False) #False if request rights are revoked
+    recieved = Column(Float, default = 0.00) #How much money has this account recieved
+    num_recieved = Column(Integer, default=0) #How many individual donations has this account recieved
+    given = Column(Float, default = 0.00) #How much money has this account given
+    num_given = Column(Integer, default=0) #How many individual donations has this accouunt given
+
+    def __init__(self, username):
+        self.username = username
+        self.account_token = 'TOKEN'
```
Comment (flllra) [inline schema.py:47]: is this gonna be a token we make or one made from the single sign on?

**Review by flllra — APPROVED** [review]

File: controls.py (hunk 3 above)
Comment (HannahSMead) [reply controls.py:36]: I agree that everyone should title theirs, but I feel like this title is silly enough to push people to not forget it!

File: controls.py (hunk 2 above)
Comment (HannahSMead) [reply controls.py:20]: We should, it is now on my list

**Pull request merged**